Source/NameServer/AddressBook/AddressBook_Task.cpp

```diff
@@ -0,0 +1,887 @@
+                       std::string &E, TaskState_t S, unsigned long DC,
+                       unsigned long EC)
+{
+    Name = N;
+    Path = P;
+    XML = X;
+    ExecutablePath = E;
+    State = Loaded;
+    DeviceCount = DC;
+    DeviceCountLd = 0;
+    ExternalCount = EC;
+    ExternalCountLd = 0;
+    SupervisorCount = 0;
+
+    State = S;
+}
+
+/*==============================================================================
+ * TaskData_t::size(): Return the size (in bytes) of the address record.
+ *============================================================================*/
+int TaskData_t::size() const
+{
+    int size = 0;
+    
+    size += sizeof(unsigned long)*5;      // Size of the 5 longs
+    size += sizeof(char) * Name.size();   // Size of the name
+    size += sizeof(char) * Path.size();   // Size of the path
+    size += sizeof(char) * XML.size();    // Size of the XML Filename
+    size += sizeof(char) * ExecutablePath.size();   // Size of the exec path
+    size += sizeof(TaskState_t);
+    
+    for(std::vector<DevTypeRecord_t>::const_iterator DT = DeviceTypes.begin();
+        DT != DeviceTypes.end(); DT++)    // Size of the device types
+    {
+        size += DT->size();
+    }
+    
+    for(std::vector<std::string>::const_iterator MT = MessageTypes.begin();
+        MT != MessageTypes.end(); MT++)   // Size of the Message Types
+    {
+        size += MT->size();
+    }
+    
+    for(std::vector<std::string>::const_iterator AT = AttributeTypes.begin();
+        AT != AttributeTypes.end(); AT++) // Size of the Attribute Types
+    {
+        size += AT->size();
+    }
+    
+    return size;
+}
+
+/*==============================================================================
+ * TaskRecord - stores data about the task in an instance of AddressBook
+ *============================================================================*/
+TaskRecord_t::TaskRecord_t()
+{
+    Name = "";
+    Path = "";
+    XML = "";
+    ExecPath = "";
+    DevCntMax = 0;
+    ExtCntMax = 0;
+    DevCnt = 0;
+    SupCnt = 0;
+    ExtCnt = 0;
+
+    State = Loaded;
+
+    TaskValid = true;
+    MapValid = true;
+    LinkValid = true;
+}
+
+/*==============================================================================
+ * TaskRecord_t::TaskRecord_t(): Full initialiser for a TaskRecord_t
+ *============================================================================*/
+TaskRecord_t::TaskRecord_t(std::string &N, std::string &P, std::string &X,
+                           std::string &E, TaskState_t S, unsigned long DC, 
+                           unsigned long EC)
+{
+    Name = N;
+    Path = P;
+    XML = X;
+    ExecPath = E;
+    DevCntMax = DC;
+    ExtCntMax = EC;
+    DevCnt = 0;
+    SupCnt = 0;
+    ExtCnt = 0;
+
+    State = S;
+
+    TaskValid = true;
+    MapValid = true;
+    LinkValid = true;
+}
+
+
+/*==============================================================================
```

Comment:
Useful comment - having this expanded and added to the documentation would have considerable worth.

Reply:
Will add